@codex please fix

```
Run python -m pytest -q tests/e2e/test_ui.py -k 'landing_chat_real_inference_with_desktop_bridge_api_v1'
============================= test session starts ==============================
platform linux -- Python 3.11.15, pytest-8.4.2, pluggy-1.6.0
benchmark: 5.1.0 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /home/runner/work/token.place/token.place
configfile: pytest.ini
plugins: base-url-2.1.0, playwright-0.7.1, cov-7.1.0, mock-3.15.1, anyio-4.13.0, benchmark-5.1.0
collected 7 items / 6 deselected / 1 selected

tests/e2e/test_ui.py::test_landing_chat_real_inference_with_desktop_bridge_api_v1 FAILED [100%]

=================================== FAILURES ===================================
_________ test_landing_chat_real_inference_with_desktop_bridge_api_v1 __________

page = <Page url='http://localhost:5010/'>, base_url = 'http://localhost:5010'
setup_servers = (<Popen: returncode: None args: ['/opt/hostedtoolcache/Python/3.11.15/x64/bin...>, None)

    @pytest.mark.e2e
    def test_landing_chat_real_inference_with_desktop_bridge_api_v1(
        page: Page,
        base_url: str,
        setup_servers,
    ):
        """
        Validate relay landing chat end-to-end through the desktop compute-node bridge.
    
        This test intentionally avoids route mocking so it verifies the real wiring:
        browser UI -> relay.py API v1 -> relay sink/source -> desktop bridge runtime.
        """
        relay_process, _ = setup_servers
        assert relay_process is not None
    
        test_env = os.environ.copy()
        test_env["TOKEN_PLACE_ENV"] = "testing"
        test_env["USE_MOCK_LLM"] = "0"
        preprovisioned_model_path = os.environ.get("TOKENPLACE_REAL_E2E_MODEL_PATH", "").strip()
        if not preprovisioned_model_path:
            raise AssertionError(
                "TOKENPLACE_REAL_E2E_MODEL_PATH must be configured for the always-on relay landing-page real-inference guardrail."
            )
        if not os.path.isfile(preprovisioned_model_path):
            raise AssertionError(
                "TOKENPLACE_REAL_E2E_MODEL_PATH must point to an existing model file "
                f"(got: {preprovisioned_model_path})."
            )
    
        bridge_process = subprocess.Popen(
            [
                sys.executable,
                "desktop-tauri/src-tauri/python/compute_node_bridge.py",
                "--model",
                preprovisioned_model_path,
                "--mode",
                "cpu",
                "--relay-url",
                base_url,
            ],
            stdin=subprocess.PIPE,
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
            text=True,
            env=test_env,
        )
    
        stdout_queue: "queue.Queue[str]" = queue.Queue()
        stderr_lines: list[str] = []
    
        def _drain_stream(stream, output_queue=None, collector=None):
            for stream_line in iter(stream.readline, ""):
                if output_queue is not None:
                    output_queue.put(stream_line)
                if collector is not None:
                    collector.append(stream_line)
    
        stdout_thread = threading.Thread(
            target=_drain_stream,
            args=(bridge_process.stdout, stdout_queue, None),
            daemon=True,
        )
        stderr_thread = threading.Thread(
            target=_drain_stream,
            args=(bridge_process.stderr, None, stderr_lines),
            daemon=True,
        )
        stdout_thread.start()
        stderr_thread.start()
    
        v1_requests = []
        v2_requests = []
    
        def record_v1_request(route):
            v1_requests.append(route.request)
            route.continue_()
    
        def record_v2_request(route):
            v2_requests.append(route.request.url)
            route.continue_()
    
        page.route("**/api/v1/chat/completions", record_v1_request)
        page.route("**/api/v2/chat/completions", record_v2_request)
    
        v1_response_headers = []
    
        def record_v1_response(response):
            if "/api/v1/chat/completions" in response.url:
                v1_response_headers.append(response.headers)
    
        page.on("response", record_v1_response)
    
        try:
            start_deadline = time.time() + 25
            registered = False
            started = False
            runtime_supports_real_inference = False
    
            while time.time() < start_deadline:
                try:
                    line = stdout_queue.get(timeout=0.5)
                except queue.Empty:
                    if bridge_process.poll() is not None:
                        stderr_output = "".join(stderr_lines)
                        raise AssertionError(
                            f"desktop bridge exited early rc={bridge_process.returncode}\n{stderr_output}"
                        )
                    continue
    
                try:
                    payload = json.loads(line)
                except json.JSONDecodeError:
                    continue
                event_type = payload.get("type")
    
                if event_type == "started":
                    started = bool(payload.get("running"))
                    assert payload.get("use_mock_llm") is False
                    assert payload.get("llama_repo_stub_imported") is False
                    llama_module_path = payload.get("llama_module_path", "")
                    assert isinstance(llama_module_path, str) and llama_module_path
                    assert not llama_module_path.endswith("/llama_cpp.py")
                    assert not llama_module_path.endswith("\\llama_cpp.py")
                    runtime_supports_real_inference = llama_module_path != "missing"
                if event_type == "status" and payload.get("registered") is True:
                    registered = True
                    break
                if event_type == "error":
                    raise AssertionError(f"desktop bridge error event: {payload}")
    
            assert started, "desktop bridge did not emit a started event"
            assert registered, "desktop bridge never reported relay registration"
            relay_ready = False
            relay_next_server_body = ""
            for _ in range(20):
                next_server_response = page.request.get(f"{base_url}/next_server")
                if next_server_response.ok:
                    relay_next_server_body = next_server_response.text()
                    try:
                        payload = next_server_response.json()
                    except Exception:  # pragma: no cover - defensive for non-json relay errors
                        payload = {}
                    if isinstance(payload, dict) and payload.get("server_public_key"):
                        relay_ready = True
                        break
                time.sleep(0.25)
            assert relay_ready, (
                "desktop bridge reported registered but relay /next_server did not expose "
                f"an active server_public_key in time. Last response body: {relay_next_server_body!r}"
            )
    
            page.goto(base_url)
            page.wait_for_load_state("networkidle")
    
            textarea = page.locator("textarea").first
            textarea.fill("What is the capital of France? Respond with one word.")
            page.evaluate(
                """
                () => {
                    window.__assistantTextSnapshots = [];
                    const container = document.querySelector('.chat-container');
                    if (!container) {
                        return;
                    }
                    const observer = new MutationObserver(() => {
                        const nodes = document.querySelectorAll('.assistant-message');
                        if (!nodes.length) return;
                        const latest = nodes[nodes.length - 1];
                        const text = (latest.textContent || '').trim();
                        window.__assistantTextSnapshots.push(text);
                    });
                    observer.observe(container, { childList: true, subtree: true, characterData: true });
                    window.__assistantObserver = observer;
                }
                """
            )
            page.locator("button", has_text="Send").click()
    
            assistant_message = page.locator(".assistant-message").last
            assistant_message.wait_for(state="visible")
    
            page.wait_for_function(
                """
                ({ selector }) => {
                    const nodes = document.querySelectorAll(selector);
                    if (!nodes.length) return false;
                    const latest = nodes[nodes.length - 1];
                    return Boolean(latest.textContent && latest.textContent.trim().length > 0);
                }
                """,
                arg={
                    "selector": ".assistant-message",
                },
            )
    
            assistant_text = assistant_message.inner_text()
            assert assistant_text.strip(), "assistant response should not be empty"
            assert "Sorry, I encountered an issue generating a response." not in assistant_text
            assert "Unknown streaming error" not in assistant_text
    
            assert len(v1_requests) >= 1
            assert v2_requests == []
            assert v1_response_headers, "expected at least one API v1 response"
            latest_headers = v1_response_headers[-1]
            provider_class = latest_headers.get("x-tokenplace-api-v1-provider")
            stream_mode = latest_headers.get("x-tokenplace-api-v1-stream-mode")
            resolved_provider_path = latest_headers.get("x-tokenplace-api-v1-resolved-provider-path")
            execution_backend_path = latest_headers.get("x-tokenplace-api-v1-execution-backend-path")
            provider_diagnostics = (
                "landing-page real-provider guardrail diagnostics: "
                f"provider_class={provider_class!r}, resolved_provider_path={resolved_provider_path!r}, "
                f"execution_backend_path={execution_backend_path!r}, stream_mode={stream_mode!r}; "
                "expected provider_class='DistributedApiV1ComputeProvider', "
                "resolved_provider_path='distributed', "
                "execution_backend_path='registered_desktop_compute_node', "
                "stream_mode='non-streaming'"
            )
            assert provider_class == "DistributedApiV1ComputeProvider", provider_diagnostics
            assert stream_mode == "non-streaming", provider_diagnostics
            assert resolved_provider_path == "distributed", (
                "landing-page real-provider guardrail requires resolved provider path "
                f"'distributed'. {provider_diagnostics}"
            )
            assert execution_backend_path == "registered_desktop_compute_node", (
                "landing-page real-provider guardrail requires execution backend path "
                f"'registered_desktop_compute_node'. {provider_diagnostics}"
            )
            if runtime_supports_real_inference:
                assert assistant_text.strip().lower() != "stub", (
                    "assistant response must not be stub when runtime reports real inference support. "
                    f"{provider_diagnostics}"
                )
    
            page.wait_for_timeout(300)
            non_streaming_state = page.evaluate(
                """
                () => {
                    if (window.__assistantObserver) {
                        window.__assistantObserver.disconnect();
                    }
                    const snapshots = Array.isArray(window.__assistantTextSnapshots)
                        ? window.__assistantTextSnapshots
                        : [];
                    const nonEmpty = snapshots.filter((value) => typeof value === 'string' && value.length > 0);
                    const uniqueNonEmpty = [...new Set(nonEmpty)];
    
                    const appEl = document.querySelector('#app');
                    const vm = appEl && appEl.__vue__;
                    const history = vm && Array.isArray(vm.chatHistory) ? vm.chatHistory : [];
                    const assistant = [...history].reverse().find((message) => message && message.role === 'assistant') || null;
    
                    return {
                        snapshots,
                        uniqueNonEmpty,
                        hasAssistant: Boolean(assistant),
                        hasDisplayContent: Boolean(
                            assistant && Object.prototype.hasOwnProperty.call(assistant, 'displayContent')
                        ),
                        hasIsTyping: Boolean(
                            assistant && Object.prototype.hasOwnProperty.call(assistant, 'isTyping')
                        ),
                        assistantIsTyping: Boolean(assistant && assistant.isTyping),
                        assistantContent: assistant && typeof assistant.content === 'string' ? assistant.content : '',
                        domAssistantText: (() => {
                            const nodes = document.querySelectorAll('.assistant-message');
                            if (!nodes.length) {
                                return '';
                            }
                            const latest = nodes[nodes.length - 1];
                            return (latest.textContent || '').trim();
                        })(),
                    };
                }
                """
            )
            assert non_streaming_state["hasAssistant"] is True
            assert non_streaming_state["hasDisplayContent"] is False
            assert non_streaming_state["assistantIsTyping"] is False
            assert len(non_streaming_state["uniqueNonEmpty"]) == 1, (
                "assistant message should render atomically without multi-step text growth; "
                f"snapshots={non_streaming_state['snapshots']}"
            )
>           assert non_streaming_state["assistantContent"].strip() == non_streaming_state["domAssistantText"].strip(), (
                "final assistant Vue state content must exactly match rendered DOM text to prove final "
                "non-streaming rendering path"
            )
E           AssertionError: final assistant Vue state content must exactly match rendered DOM text to prove final non-streaming rendering path
E           assert 'Later, after...ne day, "Good' == 'Later, after...ne day, "Good'
E             
E             - Later, after a knock on the door. Her mom opened the door and said, "Lets are very expensive, but you can't have them now".Lucushed Lena, her eyes wide with excitement, and thanked her mom for the idea. She ran inside and began to unpack the items.Lena's mom smiled and said, "Now I know what an expensive thing is!"Lena was so excited that she ran around the house and told all her mom, "I understand! I'm going to be able to get lots of money from those weighs. I want to go on this trip and share what I learned".Lena's now understands of money and happier things. She realized that she was lucky to get a job of helping in an emergency! Once upon a time, there was a little boy who was always very curious. He never talked to his mom or dad before his mom.One day, Mom asked his mom, "Do you want me to go for a drive?" But her daughter said, "No, I want to go for dinner time." Mom thought to herself and said, "No, can' have a conversation in her friend'mth her room. She said, "Okay!Mom was so happy, and she gave her mom ate away. After her mom went to her and decided to come out the to her daughter'veals, told her to his mum, which he was so his parents that he had a big eyes and said."That was very determined."You: on the to a patience."M: "P inside of his house. He replied and said to the to the chest and ate his heart. He heard a house.Lets."Lus heard her door."Dull. ancing."Fast'd the door. She.The door on the to come from inside.They said.They. She was so many times to her to the door.The door.The door!"Hif on her mom'berries. She looked to her parents'als is her and the doors.One toe. 10 loose.One of life.Oneways.He looked for a bit of the door in life on her feet.One minute.Oneways. She headsh. Sheighed down the news. She. She said, butter and her windowens. She washing down. Sheet.A thanks to her little money.Mater.M when it washing on her Mom'sebrowsyekases from then.Mater enough. icy. Sheypts. She washing on her mommy. Shebs from herbs inchty. Herballs and she had a minute.Mommy, and cooking herbs.Maring her family's.Making and it!Herbft. Shebs she wouldyobas to the door to her mommy's.Mommy, butter!Mine! Shears. Shears. Shebey, and Mommy. Sheep-ure of his story to help. Mommy."Can healals. Sheep-ure and they went on her mommy she had a different kinds of her mommy Samanthalalal-priggy hairyph-door. Shear, so late.Her mom'veak to deallessly, Sally and she said. It. Sheunry-vekey and Sammy farm. Sheep toast and hearing."Why of the city people."You, thank her friends. It was a few friends.Inst each three children whomy, and behray toothers, and wrink, butend, who Once upon her who was like to the tell him and the best of her name and she would welcome to her parents, butcherout honey and downicicic.One day was toys.The end. It'llion, ohimals. One day! Once it, and beh later! The next time to her own family.One weeking pendant, to her family'dess.A few death, and sheal. Mom and she’s.Less door. Lily. The same, she could Mimating day of life.Less was obed infectionstead, Lung, butter, for dinner dinner time and Lance. She was.M Saturday. Once it was now.One day, "Good
E             + Later, after a knock on the door. Her mom opened the door and said, "Lets are very expensive, but you can't have them now".
E             + Lucushed Lena, her eyes wide with excitement, and thanked her mom for the idea. She ran inside and began to unpack the items.
E             + Lena's mom smiled and said, "Now I know what an expensive thing is!"
E             + Lena was so excited that she ran around the house and told all her mom, "I understand! I'm going to be able to get lots of money from those weighs. I want to go on this trip and share what I learned".
E             + Lena's now understands of money and happier things. She realized that she was lucky to get a job of helping in an emergency! Once upon a time, there was a little boy who was always very curious. He never talked to his mom or dad before his mom.
E             + One day, Mom asked his mom, "Do you want me to go for a drive?" But her daughter said, "No, I want to go for dinner time." Mom thought to herself and said, "No, can' have a conversation in her friend'mth her room. She said, "Okay!
E             + Mom was so happy, and she gave her mom ate away. After her mom went to her and decided to come out the to her daughter'veals, told her to his mum, which he was so his parents that he had a big eyes and said.
E             + "That was very determined.
E             + "You: on the to a patience.
E             + "M: "P inside of his house. He replied and said to the to the chest and ate his heart. He heard a house.
E             + Lets.
E             + "Lus heard her door.
E             + "Dull. ancing.
E             + "Fast'd the door. She.
E             + The door on the to come from inside.
E             + They said.
E             + They. She was so many times to her to the door.
E             + The door.
E             + The door!
E             + "Hif on her mom'berries. She looked to her parents'als is her and the doors.
E             + One toe. 10 loose.
E             + One of life.
E             + Oneways.
E             + He looked for a bit of the door in life on her feet.
E             + One minute.
E             + Oneways. She headsh. Sheighed down the news. She. She said, butter and her windowens. She washing down. Sheet.
E             + A thanks to her little money.
E             + Mater.
E             + M when it washing on her Mom'sebrowsyekases from then.
E             + Mater enough. icy. Sheypts. She washing on her mommy. Shebs from herbs inchty. Herballs and she had a minute.
E             + Mommy, and cooking herbs.
E             + Maring her family's.
E             + Making and it!
E             + Herbft. Shebs she wouldyobas to the door to her mommy's.
E             + Mommy, butter!
E             + Mine! Shears. Shears. Shebey, and Mommy. Sheep-ure of his story to help. Mommy.
E             + "Can healals. Sheep-ure and they went on her mommy she had a different kinds of her mommy Samanthalalal-priggy hairyph-door. Shear, so late.
E             + Her mom'veak to deallessly, Sally and she said. It. Sheunry-vekey and Sammy farm. Sheep toast and hearing.
E             + "Why of the city people.
E             + "You, thank her friends. It was a few friends.
E             + Inst each three children whomy, and behray toothers, and wrink, butend, who Once upon her who was like to the tell him and the best of her name and she would welcome to her parents, butcherout honey and downicicic.
E             + One day was toys.
E             + The end. It'llion, ohimals. One day! Once it, and beh later! The next time to her own family.
E             + One weeking pendant, to her family'dess.
E             + A few death, and sheal. Mom and she’s.
E             + Less door. Lily. The same, she could Mimating day of life.
E             + Less was obed infectionstead, Lung, butter, for dinner dinner time and Lance. She was.
E             + M Saturday. Once it was now.
E             + One day, "Good

tests/e2e/test_ui.py:589: AssertionError
---------------------------- Captured stdout setup -----------------------------

Setting up servers for E2E tests...
Started relay server on port 5010 with real-provider guardrails
✓ Relay server is running
Real desktop-bridge guardrail mode: skipping server.py registration bootstrap to avoid non-bridge node selection
----------------------------- Captured stdout call -----------------------------
Browser Console [info]: <redacted>
Browser Console [info]: <redacted>
=========================== short test summary info ============================
FAILED tests/e2e/test_ui.py::test_landing_chat_real_inference_with_desktop_bridge_api_v1 - AssertionError: final assistant Vue state content must exactly match rendered DOM text to prove final non-streaming rendering path
assert 'Later, after...ne day, "Good' == 'Later, after...ne day, "Good'
  
  - Later, after a knock on the door. Her mom opened the door and said, "Lets are very expensive, but you can't have them now".Lucushed Lena, her eyes wide with excitement, and thanked her mom for the idea. She ran inside and began to unpack the items.Lena's mom smiled and said, "Now I know what an expensive thing is!"Lena was so excited that she ran around the house and told all her mom, "I understand! I'm going to be able to get lots of money from those weighs. I want to go on this trip and share what I learned".Lena's now understands of money and happier things. She realized that she was lucky to get a job of helping in an emergency! Once upon a time, there was a little boy who was always very curious. He never talked to his mom or dad before his mom.One day, Mom asked his mom, "Do you want me to go for a drive?" But her daughter said, "No, I want to go for dinner time." Mom thought to herself and said, "No, can' have a conversation in her friend'mth her room. She said, "Okay!Mom was so happy, and she gave her mom ate away. After her mom went to her and decided to come out the to her daughter'veals, told her to his mum, which he was so his parents that he had a big eyes and said."That was very determined."You: on the to a patience."M: "P inside of his house. He replied and said to the to the chest and ate his heart. He heard a house.Lets."Lus heard her door."Dull. ancing."Fast'd the door. She.The door on the to come from inside.They said.They. She was so many times to her to the door.The door.The door!"Hif on her mom'berries. She looked to her parents'als is her and the doors.One toe. 10 loose.One of life.Oneways.He looked for a bit of the door in life on her feet.One minute.Oneways. She headsh. Sheighed down the news. She. She said, butter and her windowens. She washing down. Sheet.A thanks to her little money.Mater.M when it washing on her Mom'sebrowsyekases from then.Mater enough. icy. Sheypts. She washing on her mommy. Shebs from herbs inchty. Herballs and she had a minute.Mommy, and cooking herbs.Maring her family's.Making and it!Herbft. Shebs she wouldyobas to the door to her mommy's.Mommy, butter!Mine! Shears. Shears. Shebey, and Mommy. Sheep-ure of his story to help. Mommy."Can healals. Sheep-ure and they went on her mommy she had a different kinds of her mommy Samanthalalal-priggy hairyph-door. Shear, so late.Her mom'veak to deallessly, Sally and she said. It. Sheunry-vekey and Sammy farm. Sheep toast and hearing."Why of the city people."You, thank her friends. It was a few friends.Inst each three children whomy, and behray toothers, and wrink, butend, who Once upon her who was like to the tell him and the best of her name and she would welcome to her parents, butcherout honey and downicicic.One day was toys.The end. It'llion, ohimals. One day! Once it, and beh later! The next time to her own family.One weeking pendant, to her family'dess.A few death, and sheal. Mom and she’s.Less door. Lily. The same, she could Mimating day of life.Less was obed infectionstead, Lung, butter, for dinner dinner time and Lance. She was.M Saturday. Once it was now.One day, "Good
  + Later, after a knock on the door. Her mom opened the door and said, "Lets are very expensive, but you can't have them now".
  + Lucushed Lena, her eyes wide with excitement, and thanked her mom for the idea. She ran inside and began to unpack the items.
  + Lena's mom smiled and said, "Now I know what an expensive thing is!"
  + Lena was so excited that she ran around the house and told all her mom, "I understand! I'm going to be able to get lots of money from those weighs. I want to go on this trip and share what I learned".
  + Lena's now understands of money and happier things. She realized that she was lucky to get a job of helping in an emergency! Once upon a time, there was a little boy who was always very curious. He never talked to his mom or dad before his mom.
  + One day, Mom asked his mom, "Do you want me to go for a drive?" But her daughter said, "No, I want to go for dinner time." Mom thought to herself and said, "No, can' have a conversation in her friend'mth her room. She said, "Okay!
  + Mom was so happy, and she gave her mom ate away. After her mom went to her and decided to come out the to her daughter'veals, told her to his mum, which he was so his parents that he had a big eyes and said.
  + "That was very determined.
  + "You: on the to a patience.
  + "M: "P inside of his house. He replied and said to the to the chest and ate his heart. He heard a house.
  + Lets.
  + "Lus heard her door.
  + "Dull. ancing.
  + "Fast'd the door. She.
  + The door on the to come from inside.
  + They said.
  + They. She was so many times to her to the door.
  + The door.
  + The door!
  + "Hif on her mom'berries. She looked to her parents'als is her and the doors.
  + One toe. 10 loose.
  + One of life.
  + Oneways.
  + He looked for a bit of the door in life on her feet.
  + One minute.
  + Oneways. She headsh. Sheighed down the news. She. She said, butter and her windowens. She washing down. Sheet.
  + A thanks to her little money.
  + Mater.
  + M when it washing on her Mom'sebrowsyekases from then.
  + Mater enough. icy. Sheypts. She washing on her mommy. Shebs from herbs inchty. Herballs and she had a minute.
  + Mommy, and cooking herbs.
  + Maring her family's.
  + Making and it!
  + Herbft. Shebs she wouldyobas to the door to her mommy's.
  + Mommy, butter!
  + Mine! Shears. Shears. Shebey, and Mommy. Sheep-ure of his story to help. Mommy.
  + "Can healals. Sheep-ure and they went on her mommy she had a different kinds of her mommy Samanthalalal-priggy hairyph-door. Shear, so late.
  + Her mom'veak to deallessly, Sally and she said. It. Sheunry-vekey and Sammy farm. Sheep toast and hearing.
  + "Why of the city people.
  + "You, thank her friends. It was a few friends.
  + Inst each three children whomy, and behray toothers, and wrink, butend, who Once upon her who was like to the tell him and the best of her name and she would welcome to her parents, butcherout honey and downicicic.
  + One day was toys.
  + The end. It'llion, ohimals. One day! Once it, and beh later! The next time to her own family.
  + One weeking pendant, to her family'dess.
  + A few death, and sheal. Mom and she’s.
  + Less door. Lily. The same, she could Mimating day of life.
  + Less was obed infectionstead, Lung, butter, for dinner dinner time and Lance. She was.
  + M Saturday. Once it was now.
  + One day, "Good
======================= 1 failed, 6 deselected in 18.99s =======================
Error: Process completed with exit code 1.
```

new codex task, not a r/e/v/i/e/w task

--- a/tests/e2e/test_ui.py
+++ b/tests/e2e/test_ui.py
@@ -591,8 +591,8 @@ def test_landing_chat_real_inference_with_desktop_bridge_api_v1(
 
         # DOM rendering may inject or remove whitespace around wrapped/newline boundaries
         # while preserving the same non-streaming payload text.
-        assistant_content_compact = "".join(assistant_content.split())
-        dom_assistant_text_compact = "".join(dom_assistant_text.split())
+        assistant_content_compact = " ".join(assistant_content.split())
+        dom_assistant_text_compact = " ".join(dom_assistant_text.split())
         assert assistant_content_compact == dom_assistant_text_compact, (
             "final assistant Vue state content must match rendered DOM text (allowing equivalent "
             "whitespace formatting) to prove final non-streaming rendering path"

--- a/tests/e2e/test_ui.py
+++ b/tests/e2e/test_ui.py
@@ -586,9 +586,15 @@ def test_landing_chat_real_inference_with_desktop_bridge_api_v1(
             "assistant message should render atomically without multi-step text growth; "
             f"snapshots={non_streaming_state['snapshots']}"
         )
-        assert non_streaming_state["assistantContent"].strip() == non_streaming_state["domAssistantText"].strip(), (
-            "final assistant Vue state content must exactly match rendered DOM text to prove final "
-            "non-streaming rendering path"
+        assistant_content = non_streaming_state["assistantContent"].strip()
+        dom_assistant_text = non_streaming_state["domAssistantText"].strip()
+
+        # DOM rendering may include line-wrap/newline formatting while preserving the same text payload.
+        assistant_content_compact = " ".join(assistant_content.split())
+        dom_assistant_text_compact = " ".join(dom_assistant_text.split())
+        assert assistant_content_compact == dom_assistant_text_compact, (
+            "final assistant Vue state content must match rendered DOM text (allowing equivalent "
+            "whitespace formatting) to prove final non-streaming rendering path"
         )
 
         encrypted_request = v1_requests[0].post_data_json

--- a/tests/e2e/test_ui.py
+++ b/tests/e2e/test_ui.py
@@ -589,9 +589,10 @@ def test_landing_chat_real_inference_with_desktop_bridge_api_v1(
         assistant_content = non_streaming_state["assistantContent"].strip()
         dom_assistant_text = non_streaming_state["domAssistantText"].strip()
 
-        # DOM rendering may include line-wrap/newline formatting while preserving the same text payload.
-        assistant_content_compact = " ".join(assistant_content.split())
-        dom_assistant_text_compact = " ".join(dom_assistant_text.split())
+        # DOM rendering may inject or remove whitespace around wrapped/newline boundaries
+        # while preserving the same non-streaming payload text.
+        assistant_content_compact = "".join(assistant_content.split())
+        dom_assistant_text_compact = "".join(dom_assistant_text.split())
         assert assistant_content_compact == dom_assistant_text_compact, (
             "final assistant Vue state content must match rendered DOM text (allowing equivalent "
             "whitespace formatting) to prove final non-streaming rendering path"

